src/render/graph.rs

```diff
@@ -142,9 +148,7 @@ impl Graph {
     }
 
     pub fn add_edge(&mut self, source: (AudioNodeId, usize), dest: (AudioNodeId, usize)) {
-        self.nodes
-            .get_mut(&source.0)
-            .unwrap_or_else(|| panic!("cannot connect {:?} to {:?}", source, dest))
+        self.nodes[source.0 .0 as usize]
```

Comment:
I'm a bit confused with this syntax: `source.0 .0 as usize`, what does it mean?

Reply:
Yeah that looks weird. It is a double tuple unpack for which rustfmt has added an extra space. This appears to be not necessary anymore if we change the config: https://github.com/rust-lang/rustfmt/issues/5745 
I will pick that up in a separate PR

Then again I agree the nested tuple is not nice to read. I will change the NodeCollection interface to take AudioNodeId instead of usize for indexing. This will make the code nicer!

Reply:
> I will change the NodeCollection interface to take AudioNodeId instead of usize for indexing. This will make the code nicer!

+1